setup.cfg

```diff
@@ -5,8 +5,9 @@ doc_files = docs/Makefile docs/source/*.rst
 
 [flake8]
 max-line-length = 120
-per-file-ignores:
-    pyresample/test/*.py:D102
+ignore = D107
+per-file-ignores =
+    pyresample/test/*.py:D101,D102,D103,D106
```

Comment:
I don't really agree here with ignoring D101 and D106. I think class docstrings need to be there as they are more general than method docstrings.

Reply:
I'm not against removing those ignores, but also how often do we have a class in a test module that isn't something like:

```python
class TestResampler:
    """Test basic functionality of the resamplers."""

class TestResamplerCache:
    """Test caching functionality of resamplers."""

```

Reply:
I'd rather have redundant docstring, that still helps make sure the class name is relevant for the reviewers :)

Reply:
> that still helps make sure the class name is relevant for the reviewers

How? If the docstring just repeats the information in the class name then how is it providing any more context to the reviewer?

Reply:
It's especially in the case where the name and docstring differ that it is interesting for the reviewers.

Reply:
> differ

Meaning they don't match or meaning that the docstring actually contains more information than the class name provided? In that case there is nothing stopping someone from adding a docstring to provide this information. This ignore on flake8 isn't saying "don't add docstrings" or "you can't add docstrings", it is just saying "if it doesn't make sense to add a docstring, we don't require it". If a reviewer says "I can't tell what this test class is doing" then their review can be "please add a docstring to explain what this test class is doing".

Reply:
> I doesn't take long to write, so why not?

Spent time for no benefit.

> I want to enforce having a docstring to make sure that as a reviewer that the name of the class matches what it actually does.

How does enforcing a docstring do this though? If I copy and paste a test class and update the class name (something I have to do to get both the original class and new class to exist in the same namespace), but forget to update the docstring then flake8 isn't going to catch this issue. Now the reviewer has to notice that this was a copy/paste issue (assuming they actually catch it) and the author has to fix it. Plus the docstring doesn't tell you if the tests match "what it actually does", the reviewer still has to look at the test to make sure that they do what the class name actually suggests they do.